src/main/scala/org/constellation/primitives/Schema.scala

```diff
@@ -28,8 +28,46 @@ object Schema {
   object NodeState extends Enumeration {
     type NodeState = Value
 
-    val PendingDownload, DownloadInProgress, DownloadCompleteAwaitingFinalSync, Ready, Leaving, Offline =
+    val PendingDownload, DownloadInProgress, DownloadCompleteAwaitingFinalSync, SnapshotCreation, Ready, Leaving,
+      Offline =
       Value
+
+    val all: Set[Schema.NodeState.Value] = values.toSet
+
+    val readyStates: Set[Schema.NodeState.Value] = Set(NodeState.Ready, NodeState.SnapshotCreation)
+
+    val initial: Set[NodeState] = Set(Offline, PendingDownload)
+
+    val broadcastStates: Set[Schema.NodeState.Value] = Set(Ready, Leaving, Offline, PendingDownload)
+
+    val validDuringDownload: Set[Schema.NodeState.Value] = Set(DownloadInProgress, DownloadCompleteAwaitingFinalSync)
+
+    val validForDownload: Set[Schema.NodeState.Value] = Set(PendingDownload, Ready)
+
+    val validForRedownload: Set[Schema.NodeState.Value] = Set(Ready)
+
+    val validForSnapshotCreation: Set[Schema.NodeState.Value] = Set(Ready, Leaving)
+
+    val validForTransactionGeneration: Set[Schema.NodeState.Value] = Set(Ready, SnapshotCreation)
+
+    val validForOwnConsensus: Set[Schema.NodeState.Value] = Set(Ready, SnapshotCreation)
+
+    val validForConsensusParticipation: Set[Schema.NodeState.Value] = Set(Ready, SnapshotCreation)
+
+    def canActAsDownloadSource(current: NodeState): Boolean = Set(Ready, SnapshotCreation, Leaving).contains(current)
```

Comment:
Would you mind extracting that set as `validForDownloadSource`? As you did for the rest